CRFModuleValidation/MasterScheduledActivityManager.swift

```diff
@@ -175,33 +175,47 @@ class MasterScheduledActivityManager: ScheduledActivityManager {
     func updateSchedules(newSchedules: [Schedule], newSections: [ScheduleSection], shouldResetNotifications: Bool) {
         DispatchQueue.main.async {
             
-            let schedulesChanged = (self.schedules != newSchedules)
-            
             // Set the new values
             self.scheduleSections = newSections
             self.schedules = newSchedules
             
             // refresh the delegate
             self.delegate?.reloadFinished(self)
             
-            // update the reminders if changed
-            if schedulesChanged && shouldResetNotifications {
+            // update the reminders
+            if shouldResetNotifications {
                 self.updateReminderNotifications()
             }
         }
     }
     
+    func scheduleItemsToRemindOf() -> [ScheduleItem] {
+        let unfinishedCardioSections = self.scheduleSections.filter({
+            ($0.items[0].taskGroup.identifier as NSString).hasPrefix("Cardio") &&
+            !$0.items[0].isCompleted
+        })
+        guard unfinishedCardioSections.count > 0 else { return [] }
+        
+        var items: [ScheduleItem] = []
+        for section in unfinishedCardioSections {
+            items.append(contentsOf: section.items)
+        }
+        
+        return items
+    }
+    
     func updateReminderNotifications() {
         
         // use dispatch async to allow the method to return and put updating reminders on the next run loop
         DispatchQueue.main.async {
             
             // Remove previous reminders
-            let identifiers = self.schedules.map({ $0.taskGroup.identifier })
+            let identifiers = self.scheduleItemsToRemindOf().map({ $0.identifier })
```

Comment:
You are only removing the schedules that have **not** been finished. What if the person does the task *before* the reminder notification fires?

Might be simpler (for the purposes of this study) to just use hardcoded identifiers and remove all.

Or to use a single identifier like "org.sagebase.CRFValidationApp" since then that reminder will be removed.

Alternatively, you could also just remove all reminders rather than trying to only remove those for which the notification is setup to fire.